client/src/controllers/AutoFieldController.ts

```diff
@@ -24,6 +24,12 @@ export class AutoFieldController extends Controller<
       );
     }
 
+    // Update form action based on selected option
+    if (this.element.type === 'select-one') {
+      const selectedOption = this.element.options[this.element.selectedIndex];
+      form.action = selectedOption.dataset.action;
+    }
+
```

Comment:
```suggestion
    if (this.syncActionValue) {
      form.action = this.element.value;
    }
```

Note: this assumes you are going to use a Stimulus value approach, which will need to be set up above as follows:

```
// ...
  HTMLInputElement | HTMLSelectElement
> {
  static values = {
    /** When true, update the form action URL to match the value provided */
    syncAction: { default: false, type: Boolean },
  };

  declare syncActionValue: boolean;
```